ethereum/datastructures/src/main/java/tech/pegasys/artemis/datastructures/event/Deposit.java

```diff
@@ -15,59 +15,57 @@
 
 import com.fasterxml.jackson.core.JsonProcessingException;
 import com.fasterxml.jackson.databind.ObjectMapper;
-import java.nio.ByteOrder;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
+import org.apache.commons.lang3.ArrayUtils;
 import org.apache.tuweni.bytes.Bytes;
 import org.apache.tuweni.bytes.Bytes32;
 import tech.pegasys.artemis.datastructures.interfaces.IRecordAdapter;
 import tech.pegasys.artemis.pow.api.DepositEvent;
 import tech.pegasys.artemis.pow.contract.DepositContract.DepositEventResponse;
 import tech.pegasys.artemis.pow.event.AbstractEvent;
 import tech.pegasys.artemis.util.bls.BLSPublicKey;
+import tech.pegasys.artemis.util.bls.BLSSignature;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
 
 public class Deposit extends AbstractEvent<DepositEventResponse>
     implements DepositEvent, IRecordAdapter {
   // processed fields
   private BLSPublicKey pubkey;
   private Bytes32 withdrawal_credentials;
-  private Bytes proof_of_possession;
+  private BLSSignature signature;
   private long amount;
+  private Bytes merkel_tree_index;
 
   private static final ObjectMapper mapper = new ObjectMapper();
 
-  // raw fields
-  private Bytes data;
-  private Bytes merkel_tree_index;
+
   private Map<String, Object> outputFieldMap = new HashMap<>();
 
   public Deposit(DepositEventResponse response) {
     super(response);
-    // raw fields
-    this.data = Bytes.wrap(response.data);
+
+    ArrayUtils.reverse(response.merkle_tree_index);
     this.merkel_tree_index = Bytes.wrap(response.merkle_tree_index);
 
-    // process fields
-    this.pubkey = BLSPublicKey.fromBytesCompressed(data.slice(0, 48).reverse());
-    this.withdrawal_credentials = Bytes32.wrap(data.slice(48, 32).reverse());
-    this.proof_of_possession = data.slice(88, 96).reverse();
-    this.amount = data.slice(80, 8).toLong(ByteOrder.LITTLE_ENDIAN);
-  }
+    ArrayUtils.reverse(response.pubkey);
```

Comment:
Just curious, do you reverse all these bytes bc Eth1 is big-endian and we use bytes as Little endian?